packages/client/src/store/rxUtils.ts

```diff
@@ -72,3 +73,24 @@ export const createSubscription = <T>(
     subscription.unsubscribe();
   };
 };
+
+/**
+ * Creates a subscription and returns a function to unsubscribe. Makes sure that
+ * only one async handler runs at the same time. If updates come in quicker than
+ * it takes for the current handler to finish, other handlers will wait.
+ *
+ * @param observable the observable to subscribe to.
+ * @param handler the async handler to call when the observable emits a value.
+ */
+export const createSafeAsyncSubscription = <T>(
```

Comment:
Is this method used somewhere?

Reply:
Not right now. I wanted to use in places like this https://github.com/GetStream/stream-video-js/blob/main/packages/client/src/Call.ts#L388-L397 where it's not safe to call `call.leave()` in a callback that's now awaited. But I don't want to cram to many changes in one PR.